### ISSUE TITLE
Move product:component mapping into external config files

### ISSUE THREAD

**Issue by jdm** [jdm/bugsahoy#11]
The goal is to keep bugsahoy as customizable as possible for other groups using Bugzilla to track work items. It would be ideal to move all definition of mappings into an external config file that could be processed at deployment time, creating the static index.html that would be served to visitors. This mapping would include:
- Sections of clickable groups and names, with parameters that indicate whether to perform a union or intersection of the results
- Corresponding bugzilla searches to perform
- Extra information boxes

In other words, everything that defines the non-universal aspects of the current UI. JSON would be a reasonable choice for the config file format, with a python script to perform the translation.
